src/Scheduler.cc

```diff
@@ -157,10 +157,16 @@ void Scheduler::set_enable_chaos(bool enable_chaos) {
    * return 1 to maximize throughput (since effectively we really only have
    * one core).
    */
+  ASSERT(current_, pretend_num_cores_ == 1);
```

Comment:
I don't think we need this assertion. Why did you add it?

Reply:
The idea was that --num-cores wins out over --chaos for setting the number of cores, and that is implemented by calling set_chaos_mode first then set_num_cores (and overwriting any effect set_chaos_mode may have had) -- which means that there's an implicit ordering dependency between the two calls.

You can remove it, or I could add a comment, or merge the two calls together to remove the ordering dependency entirely (set_chaos_and_cores?). Let me know what you'd prefer.

Reply:
Oh, sorry. Further evidence that I under-commented: this assert is how I am verifying that set_num_cores was not called before set_chaos_mode.

Reply:
I think implementing that restriction using an assert here creates a dependency in `Scheduler` on how `RecordCommand` is supposed to work, which isn't great. So please remove the assert (and fix the code formatting issue I just mentioned). Also, add a comment to `setup_session_from_flags` explaining that `num_cores` has to be set after `enable_chaos`.